src/transformers/trainer.py

```diff
@@ -305,6 +306,11 @@ def __init__(
         self.deepspeed = None
         self.is_in_train = False
 
+        if args.include_inputs_for_metrics:
+            args.include_inputs_for_metrics = True
+        else:
+            args.include_inputs_for_metrics = False
```

Comment:
The `TrainingArguments` should not be touched by the `Trainer`, I don't think this is necessary in any case.
```suggestion
```